pkgs/by-name/mi/mistral-rs/package.nix

```diff
@@ -0,0 +1,201 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+
+  # nativeBuildInputs
+  pkg-config,
+  python3,
+
+  # buildInputs
+  oniguruma,
+  openssl,
+  mkl,
+  stdenv,
+  darwin,
+
+  # env
+  fetchurl,
+
+  testers,
+  mistral-rs,
+
+  cudaPackages,
+  cudaCapability ? null,
+
+  config,
+  # one of `[ null false "cuda" "mkl" "metal" ]`
+  acceleration ? null,
+
+}:
+
+let
+  accelIsValid = builtins.elem acceleration [
+    null
+    false
+    "cuda"
+    "mkl"
+    "metal"
+  ];
+
+  cudaSupport =
+    assert accelIsValid;
+    (acceleration == "cuda") || (config.cudaSupport && acceleration == null);
+
+  minRequiredCudaCapability = "6.1"; # build fails with 6.0
+  inherit (cudaPackages.cudaFlags) cudaCapabilities;
+  cudaCapabilityString =
+    if cudaCapability == null then
+      (builtins.head (
+        (builtins.filter (cap: lib.versionAtLeast cap minRequiredCudaCapability) cudaCapabilities)
```

Comment:
@EricLBuehler it's very convenient to have you here 🙃 Do you think it could be possible to pass a _list_ of gencodes through mistral-rs' build rules to nvcc, instead of just a single architecture? In Nixpkgs the user configures the expected cuda architectures globally as a list, but @GaetanLepage's current solution is to pick the oldest of the requested architectures and hope that the PTX code is included in the output.

Reply:
The compute capability is used by the `bindgen_cuda` crate. It gives its value [to the `--architecture` flag](https://github.com/Narsil/bindgen_cuda/blob/a6b0c891be8ebefb55600d46d8a77a358adc3913/src/lib.rs#L246).

Reply:
@GaetanLepage Out of scope for the PR, but we could try burning the gencode flags into `NVCC_APPEND_FLAGS`, i.e. we'd tell nvcc to build fatbins without the rust component even noticiting

Reply:
Yeah, I think that's a good idea. Mistral.rs doesn't directly call nvcc as we delegate it to `bindgen_cuda` as @GaetanLepage pointed out. The best way is probably to use some environment variable.